runtime/src/error.c

```diff
@@ -45,19 +49,52 @@
 #include <libunwind.h>
 
 #ifdef __linux__
+#include <dlfcn.h> // for dladdr
 // We create a pipe with addr2line and try to get a line number
 // Currently the precise line number works only on linux64
-static int chpl_unwind_getLineNum(char *process_name, void *addr){
-  char buf[128];
+static int chpl_unwind_getLineNum(void *addr){
+
+  // Compute the object containing the address
+  int rc;
+  Dl_info info;
+  intptr_t relativeAddr;
+  char process_name[256];
```

Comment:
The `readlink()` is going to put a path into this buffer.  256 seems a bit small for the length.  I won't send you down the `PATH_MAX` rathole; maybe just use 1k?